src/Native/configure.cmake

```diff
@@ -317,6 +317,14 @@ check_symbol_exists(
     "sys/syslimits.h"
     HAVE_OPEN_MAX)
 
+if (CMAKE_SYSTEM_NAME STREQUAL Linux)
```

Comment:
I may not have all the context here but WinCNG is unrelated to heimntlm. Anyway, heimntlm should be built on all platforms that Heimdal supports (note though that the version of Heimdal that ships with OS X builds this as /System/Library/PrivateFrameworks/CommonAuth.framework, but you shouldn't be using that as it's a private interface).